docs/source/docker/advanced/datasource_predictions.rst

```diff
@@ -0,0 +1,347 @@
+    the Lightly docker will fail! See below how to create a good `schema.json` file.
+
+
+Prediction Schema
+-----------------
+For Lightly it's required to store a prediction schema. The schema helps the Lightly
+Platform to correctly identify and display classes. It also helps to prevent errors
+as all predictions which are loaded are validated against this schema.
+
+
+For classification and object detection the prediction schema must include all the categories and ids.
+For other tasks such as keypoint detection it can be useful to store additional information
+like edges between keypoints.
+
+You can provide all this information to Lightly by adding a `schema.json` to the directory of the respective task.
+
+The schema.json file must have a key `categories`` with a corresponding list of categories following the COCO annotation format.
+It must also have a key `task_description` indicating the type of predictions. The `task_description` must be one of:
+
+ - classification
+ - object-detection
+
+
+For example, let's say we are working with a classification model predicting the weather on an image.
+The three classes are sunny, clouded, and rainy.
+
+
+.. code-block:: javascript
+    :caption: .lightly/predictions/classification_weather/schema.json
+
+    {
+        "task_description": "classification",
+        "categories": [
+            {
+                "id": 0,
+                "name": "sunny",
+            },
+            {
+                "id": 1,
+                "name": "clouded",
+            },
+            {
+                "id": 2,
+                "name": "rainy",
+            }
+        ]
+    }
+
+
+
+Prediction Files
+----------------
+Lightly requires a **single prediction file per image**. The file should be a .json
+following the format defined under :ref:`Prediction Format` and stored in the subdirectory
+`.lightly/predictions/${TASK_NAME}` in the storage bucket the dataset was configured with.
+In order to make sure Lightly can match the predictions to the correct source image,
+it's necessary to follow the naming convention:
+
+.. code-block:: bash
+
+    # filename of the prediction for image FILENAME.EXT
+    .lightly/predictions/${TASK_NAME}/${FILENAME}.json
+
+    # example: my_image.png, classification
+    .lightly/predictions/my_classification_task/my_image.json
+
+    # example: my_subdir/my_image.png, classification
+    .lightly/predictions/my_classification_task/my_subdir/my_image.json
+
+
+Prediction Files for Videos
+---------------------------
+When working with videos, Lightly requires a prediction file per frame. Lightly
+uses a naming convention to identify frames: The filename of a frame consists of
+the video filename, the video format, and the frame number (padded to the length
+of the number of frames in the video) separated by hyphens:
+
+.. code-block:: bash
+
+    # filename of the predictions of the Xth frame of video FILENAME.EXT
+    # with 200 frames (padding: len(str(200)) = 3)
```

Comment:
Maybe we can make this easier and take the example of how we calculate the padding out of this code block.
And I think it's good here (in the text above) to give 2-3 examples.
E.g. a video with 200 frames = 3 digits padding. A video with 1000 frames would have 4 digits?